dm/syncer/data_validator.go

```diff
@@ -737,8 +736,13 @@ func (v *DataValidator) stopInner() {
 	v.fromDB.Close()
 	v.toDB.Close()
 
+	// release the lock so that the error routine can process errors
+	// wait until all errors are recorded
+	v.Unlock()
 	v.wg.Wait()
```

Comment:
a return branch in line 731.

OK maybe we should be aware of nested synchronization primitives, like read/write to channel while holding the lock.

Reply:
b6d2594